src/main/pages/setup-openshift/openshift-admin-guide.md

```diff
@@ -9,3 +9,76 @@ folder: setup-openshift
+fall back to the dedicated OpenShift account configured for the Kubernetes infrastructure, as described in the [AdminGuide](admin-guide#who-creates-workspace-objects).
+
+To easily install Che on OpenShift with this feature enabled, see [this section for Minishift](openshift-multi-user#creating-workspace-resources-in-personal-openshift-accounts-on-minishift)
+and [this one for OCP](openshift-multi-user#creating-workspace-resources-in-personal-openshift-accounts)
+
+#### OpenShift identity provider registration
+
+The Keycloak OpenShift identity provider is described in [this documentation](https://www.keycloak.org/docs/3.3/server_admin/topics/identity-broker/social/openshift.html).
+
+1. In the [Keycloak administration console](user-management#auth-and-user-management), when adding the OpenShift identity provider, you should use the following settings:
+
+{% include image.html file="keycloak/openshift_identity_provider.png" %}
+
+`Base URL` is the URL of the OpenShift console
+
+2. Next thing is to add a default read-token role:
+
+{% include image.html file="git/kc_roles.png" %}
+
+3. Then this identity provider has to be declared as an OAuth client inside OpenShift. This can be done with the corresponding command:
+
+```bash
+oc create -f <(echo '
+apiVersion: v1
+kind: OAuthClient
+metadata:
+  name: kc-client
+secret: "<value set for the 'Client Secret' field in step 1>"
+redirectURIs:
+  - "<value provided in the 'Redirect URI' field in step 1>"
+grantMethod: prompt
+')
+```
+
+__Note__: Adding a OAuth client requires cluster-wide admin rights. 
+ 
+
+#### Che configuration
+
+On the Che deployment configuration:
+- the `CHE_INFRA_OPENSHIFT_PROJECT` environment variable should be set to `NULL` to ensure a new distinct OpenShift namespace is created for every started workspace.
+- the `CHE_INFRA_OPENSHIFT_OAUTH__IDENTITY__PROVIDER` environment variable should be set to the alias of the OpenShift identity provider
+specified in step 1 of its [registration in Keycloak](#openshift-identity-provider-registration). The default value is `openshift-v3`.
+
+#### Providing the OpenShift certificate to Keycloak
+
+If the certificate used by the OpenShift console is self-signed or is not trusted, then by default the Keycloak will not be able to
+contact the OpenShift console to retrieve linked tokens.
+
+In this case the OpenShift console certificate should be passed to the Keycloak deployment as an additional environment property.
+This will enable the Keycloak server to add it to its list of trusted certificates, and will fix the problem.
+
+The environment variable is named `OPENSHIFT_IDENTITY_PROVIDER_CERTIFICATE`.
+
+Since adding a multi-line certificate content in a deployment configuration environment variable is not that easy, the best way is to use a secret that contains the certificate,
+and refer to it in the environment variable.
```

Comment:
Should we put an example of how to do that?

Reply:
It's in the example for MiniShift. There's a yaml with for the secret